cli/exchange/pattern.go

```diff
@@ -192,7 +192,11 @@ func PatternPublish(org, userPw, jsonFilePath, keyFilePath, pubKeyFilePath, patN
 		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, "the org specified in the input file (%s) must match the org specified on the command line (%s)", patFile.Org, org)
 	}
 	patInput := PatternInput{Label: patFile.Label, Description: patFile.Description, Public: patFile.Public, AgreementProtocols: patFile.AgreementProtocols, UserInput: patFile.UserInput}
-
+	
+	//issue 924: Patterns with no services are not allowed 
+	if patFile.Services == nil || len(patFile.Services) == 0 {
```

Comment:
I'm not sure you are working with the latest code....I dont see msgPrinter function calls that Ling added).

Reply:
you are right, I rebased this branch with master, it should work now

Reply:
Verifying agreements: agreements still present
Sleeping for 30s giving agreements time to fail before checking again
Verifying agreements: agreements still present
Node unconfig setting is 
Unconfig loop tests are disabled.
All tests SUCCESSFUL
make: Leaving directory `/home/travis/gopath/src/github.com/open-horizon/anax/test'
The command "if [ "${TRAVIS_PULL_REQUEST}" = "false" ]; then make test -C test TEST_VARS="NOLOOP=1 TEST_PATTERNS=sall APPSTORE=1"; else make test -C test TEST_VARS="NOLOOP=1 TEST_PATTERNS=sall" && make test -C test TEST_VARS="NOLOOP=1"; fi" exited with 0.
Done. Your build exited with 0.